src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -0,0 +1,479 @@
+
+    def getXParent(self):
+        return self._xParent
+
+    def getYParent(self):
+        return self._yParent
+
+    def _addXFile(self, filename, curve=None):
+        fileItem, iconItem, removeItem = self._createRowItems(filename, curve)
+        fileItem.setData(qt.QSize(0, 30), qt.Qt.SizeHintRole)
+
+        xIndex = self._findRowContainingText("X")
+        if xIndex is not None:
+            self.setItem(xIndex, 1, fileItem)
+            self.setItem(xIndex, 2, removeItem)
+
+    def _addYFile(self, filename, curve=None, node="X"):
+        fileItem, iconItem, removeItem = self._createRowItems(filename, curve)
+        if not filename:
+            fileItem.setData(qt.QSize(0, 30), qt.Qt.SizeHintRole)
+
+        if self.getYParent().rowCount() == 0:
+            self.getYParent().appendRow([qt.QStandardItem(), fileItem])
+            return
+
+        self.getYParent().insertRow(
+            self.getYParent().rowCount() - 1, [iconItem, fileItem, removeItem]
+        )
+
+    def _findRowContainingText(self, text):
+        for row in range(self.rowCount()):
+            item = self.item(row, 0)
+            if item and item.text() == text:
+                return row
+        return None
+
+    def _createRowItems(self, filename, curve):
+        fileItem = qt.QStandardItem(filename)
+        fileItem.setData(curve, qt.Qt.UserRole)
+        fileItem.setData(False, _DROP_HIGHLIGHT_ROLE)
+        fileItem.setEditable(False)
+        fileItem.setDropEnabled(True)
+
+        iconItem = qt.QStandardItem()
+        removeItem = qt.QStandardItem()
+        iconItem.setEditable(False)
+
+        return fileItem, iconItem, removeItem
+
+    def fileExists(self, filename):
+        for row in range(self.getXParent().rowCount()):
+            item = self.item(row, 1)
+            if item and item.text() == filename:
+                return True
+        return False
+
+    def addUrl(self, url, node="X"):
+        if url.file_path() is not None:
+            file = silx.io.open(url.file_path())
+            data = file[url.data_path()]
+            if silx.io.is_dataset(data) and data.ndim == 1:
+                if node == "X":
+                    self._xDataset = data
+
+                    for item in self._plot1D.getItems():
+                        y = item.getInfo()
+                        if y is None:
+                            continue
+                        length = min(len(self._xDataset), len(y))
+                        item.setData(self._xDataset[:length], y[:length])
+
+                    self._addXFile(url.data_path())
+                else:
+                    if data is None:
+                        return
+                    curve = self._addPlot(self._xDataset, data)
+                    self._addYFile(url.data_path(), curve)
+
+    def _addPlot(self, x, y):
+        if x is None:
+            x = numpy.arange(len(y))
+
+        legend_name = f"Curve {self._index}"
```

Comment:
```suggestion
        legend = f"Curve {self._index}"
```
If you accept the modification be careful because it won't solve usage in the next lines.